### ISSUE TITLE
Updating Images/Containers?

### ISSUE THREAD

**Issue by dlover25** [MathiasFurenes/synology-arr-guide#3]
Not really an issue... 

First, this guide was super helpful for me. I had been using Download Station forever on Synology, mostly because it worked and the arr stack just seemed like too much work. This guide was great! I've got everything working very well. And these arr apps are absolutely amazing. I've done everything in your guide except for the glueTUN stuff. I had already solved the vpn thing by spinning up a vdsm image in my virtualization manager, adding a vpn connection to the network config (Nord). Using your guide, adding container manager with the arr-stack in vdsm worked great. In my main router, I added a kill switch firewall rule that restricted 100% of the VDSM traffic out the wan unless it was directed to the Nord VPN addresses that I was using. So, in case the nordvpn connection ever dropped, it wouldn't let any traffic from vdsm (and download station AND the arr-stack in container manager) out my WAN. Yes, I'd love to move everything out to the main dsm, and use glueTUN, and get rid of the VDSM image, but that seems like a bigger project than I'm willing to take on at the moment. But again thank you for your guide. Very helpful!!!

To my actual question... How do I safely update my images/containers in Container Manager. I've read to NOT update the applications within the containers themselves or you risk corrupting the data. So, I assume that means to do an equivalent of sudo docker compose pull and then restart them. Not sure how to do that in Synology Container Manager and not exactly sure what I should do with these containers specifically? So... assuming that I'm using your guide (because I am), how do I update these apps safely????

**Comment by MathiasFurenes** [comment]
Great to hear that someone got to use my guide! 

Since we are using ":latest" for all our container images (e.g. "lscr.io/linuxserver/sonarr:latest") it will pull the latest image every time you recreate the container, if I'm not mistaken. So you could update it by just stopping the project then starting again. But you should always have a snapshot to roll back to incase of something bad happening!

The thing that would be bad is using watchtower to update autoamtically. Now, I do admit I don't know the full details as I have not read the documentation since I don't use it too much, but I think it would just shut down the containers to then update it and restart. This can cause some trouble if something has changed, maybe some files can get corrupted or something entirely else. The point is, sometimes you just need it to work, and don't want to bother with troubleshooting. But if you know there is a major update and you have the time for a potential troubleshoot, you can update manually. 

TLDR; Don't use watchtower, update manually. And ALWAYS have snapshots in case of an emergency!

**Comment by dlover25** [comment]
Starting and stopping a container generally doesn't update to the latest image. In pure Linux terms, it requires a "docker compose pull".  I've just never used Container Manager to manage docker containers. It looks like maybe it's done from the "images" tab in container manager. Since there aren't any updated images yet, mine doesn't show any updates are needed. I'll keep an eye on it to see if my images tab changes with available updates.

**Comment by HartR** [comment]
you could create a recurring user script in synology to do this safely

also, thanks so much for this guide! I tried this myself a year ago and was unsuccessful

**Comment by HartR** [comment]
![image](https://github.com/user-attachments/assets/64bde685-d4a3-4506-8529-71d8a11c7bd1)
This works for me

**Comment by MathiasFurenes** [comment]
Sorry, but I didn't notice this before today!

Thank you very much for providing this script!
